utils/post.js

```diff
@@ -334,7 +334,7 @@ const modifyNewItemAnonymity = (newItem) => {
 };
 
 async function filterFeeds(userId, feeds = [], feed_id = null, threshold = null) {
-  const results = await Promise.allSettled(
+  const results = await Promise.all(
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from `Promise.allSettled` to `Promise.all` could potentially introduce unhandled promise rejections. `Promise.allSettled` ensures that all promises are settled, whether they're fulfilled or rejected, before it resolves. On the other hand, `Promise.all` rejects as soon as one of the promises in the iterable rejects, and doesn't wait for the rest of the promises to settle.

If any promise within the `feeds.map()` function rejects, it will cause an unhandled promise rejection error with `Promise.all`. If the intention is to continue processing even if some promises fail, you should stick with `Promise.allSettled`.

Here's how you can handle errors with `Promise.all`:

```diff
- const results = await Promise.all(
+ const results = await Promise.all(
     feeds.map(async (item) => {
       try {
         const expired = isExpiredPost(item);
         if (expired) {
           // ...
         }
       } catch (error) {
         console.error(error);
         return null; // or some default value
       }
     })
   );
```

This way, if there's an error in `isExpiredPost(item)`, it will be caught and logged, and the map function will return a default value, preventing `Promise.all` from rejecting.


<!-- This is an auto-generated comment by OSS CodeRabbit -->